insights/parsers/ldif_config.py

```diff
@@ -0,0 +1,133 @@
+"""
+LDIF Configuration - file ``/etc/dirsrv/slapd-*/dse.ldif``
+==========================================================
+"""
+
+from insights import Parser, parser
+from insights.specs import Specs
+from insights.parsers import SkipException, keyword_search
+import re
+
+
+@parser(Specs.ldif_config)
+class LDIFParser(Parser, list):
+    """
+    Parse the content of the directory server configuration of the
+    ``/etc/dirsrv/slapd-*/dse.ldif`` file.
+
+    The file dse.ldif is in the LDIF format. LDIF contains multi-row records
+    where each record is identified by a ``dn:`` line ("dn" as in "distinguished
+    name") and the record's other lines are attributes. The value may be specified
+    as UTF-8 text or as base64 encoded data, or a URI may be provided to the
+    location of the attribute value.
+
+    Sample output of the 'cn=changelog5,cn=config' dn attribute block of the
+    ``/etc/dirsrv/slapd-*/dse.ldif`` file::
+
+        dn: cn=changelog5,cn=config
+        cn: changelog5
+        createTimestamp: 20201026161228Z
+        creatorsName: cn=Directory Manager
+        modifiersName: cn=Directory Manager
+        modifyTimestamp: 20201026161228Z
+        nsslapd-changelogdir: /var/lib/dirsrv/slapd-IDM-NYPD-FINEST/cldb
+        nsslapd-changelogmaxage: 7d
+        objectClass: top
+        objectClass: extensibleobject
+
+    Returns:
+
+        list: A list of dictionaries for each 'dn' attribute block of the ldif configuration.
+
+    Examples:
+
+        >>> ldif_config = sorted(ldif_config, key=lambda x: x['dn'])
+        >>> ldif_config[1]['dn']
+        'cn=changelog5,cn=config'
+        >>> ldif_config[1]['modifiersName']
+        'cn=Directory Manager'
+        >>> ldif_config[1]['modifyTimestamp']
+        '20201026161228Z'
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException('The file is empty')
+
+        m_flag = 0
+        for line in content:
+            attr_kval = {}
+            line = line.strip()
+            if not line:
+                continue
+            elif line.startswith('#'):
+                # lines beginning with # are ignored
+                continue
+            # line starts with 'dn' attribute
+            elif line.startswith('dn:'):
+                aci_kline = []
+                aci_flag = 0
+                m_flag += 1
+                # line starts with 'dn' attribute line
+                if line == 'dn:':
+                    attr_kval[line.split(':', 1)[0]] = line.split(':', 1)[1]
+                    self.append(attr_kval)
+                else:
+                    attr_kval[line.split(':', 1)[0].split(' ', 1)[0]] = line.split(':', 1)[1].split(' ', 1)[1]
+                    self.append(attr_kval)
+            # line starts with 'aci' attribute
+            elif line.startswith('aci:'):
+                aci_val = line.split(':', 1)[1].split(' ', 1)[1]
+                aci_key = line.split(':', 1)[0].split(' ', 1)[0]
```

Comment:
the `aci_key` is always `aci`, right?

Reply:
Yes, it will be always "aci" but this is required at line:124 check https://github.com/RedHatInsights/insights-core/pull/3257/files#diff-81c1c4bcd3f8db07be6c1231b319edd2a851bd28e2e007c987fa79e87ff55068R124